components/brave_wallet/browser/eth_nonce_tracker_unittest.cc

```diff
@@ -35,29 +34,39 @@ namespace brave_wallet {
 
 class EthNonceTrackerUnitTest : public testing::Test {
  public:
-  EthNonceTrackerUnitTest() {
+  EthNonceTrackerUnitTest()
+      : task_environment_(base::test::TaskEnvironment::TimeSource::MOCK_TIME) {
     shared_url_loader_factory_ =
         base::MakeRefCounted<network::WeakWrapperSharedURLLoaderFactory>(
             &url_loader_factory_);
   }
 
   void SetUp() override {
-    TestingProfile::Builder builder;
-    auto prefs =
-        std::make_unique<sync_preferences::TestingPrefServiceSyncable>();
-    RegisterUserProfilePrefs(prefs->registry());
-    builder.SetPrefService(std::move(prefs));
-    profile_ = builder.Build();
+    brave_wallet::RegisterProfilePrefs(prefs_.registry());
   }
 
-  PrefService* GetPrefs() { return profile_->GetPrefs(); }
+  PrefService* GetPrefs() { return &prefs_; }
 
   network::SharedURLLoaderFactory* shared_url_loader_factory() {
     return url_loader_factory_.GetSafeWeakWrapper().get();
   }
 
   void WaitForResponse() { task_environment_.RunUntilIdle(); }
 
+  void GetNextNonce(EthNonceTracker* tracker,
+                    const std::string& address,
+                    bool expected_success,
+                    uint256_t expected_nonce) {
+    base::RunLoop run_loop;
+    tracker->GetNextNonce(
+        address, base::BindLambdaForTesting([&](bool success, uint256_t nonce) {
+          run_loop.Quit();
```

Comment:
This usually is put at the end of lambda function.

Reply:
done